docs/src/pages/components/data-grid/getting-started/migration-v4.md

```diff
@@ -0,0 +1,432 @@
+  // Make sure to convert them to use `styled()` or `<Box />` first.
+  disableGlobal: true,
+  // Class names will receive this seed to avoid name collisions.
+  seed: 'mui-jss',
+});
+```
+
+3. Create a v5 theme with the same customizations as the v4 theme. This has to be done as the theme is not shared between different MUI Core versions.
+
+```jsx
+import { createMuiTheme as createThemeV4 } from '@material-ui/core/styles';
+import { createTheme as createThemeV5 } from '@mui/material/styles';
+
+const themeV4 = createThemeV4({
+  palette: {
+    primary: {
+      main: '#2196f3',
+    },
+  },
+});
+
+const themeV5 = createThemeV5({
+  palette: {
+    primary: {
+      main: theme.palette.primary.main,
+    },
+  },
+});
+```
+
+4. Apply the class name generator and v5 theme to the application.
+
+```jsx
+import * as React from 'react';
+import { ThemeProvider as ThemeProviderV5 } from '@mui/material/styles';
+import { ThemeProvider as ThemeProviderV4, StylesProvider } from '@material-ui/core/styles';
+
+const generateClassName = createGenerateClassName({ ... });
+const themeV4 = createThemeV4({ ... });
+const themeV5 = createThemeV5({ ... });
+
+export default function DataGridDemo() {
+  return (
+    <StylesProvider generateClassName={generateClassName}>
+      <ThemeProviderV4 theme={themeV4}>
+        <ThemeProviderV5 theme={themeV5}>
+          {/* Your component tree. */}
+        </ThemeProviderV5>
+      </ThemeProviderV4>
+    </StylesProvider>
+  );
+}
+```
+
+The following interactive demo shows how these steps tie together:
+
+{{"demo": "pages/components/data-grid/getting-started/DataGridV5WithCoreV4.js", "hideToolbar": true, "bg": true}}
+
+## Migrating MUI X from v4
+
+### CSS classes
+
+- Some CSS classes were removed
+
+  1. `.MuiDataGrid-window` was removed
+  2. `.MuiDataGrid-windowContainer` was removed
+
+- Some CSS classes were renamed
+
+  1. `.MuiDataGrid-viewport` was renamed `.MuiDataGrid-virtualScroller`
+  2. `.MuiDataGrid-dataContainer` was renamed `.MuiDataGrid-virtualScrollerContent`
+  3. `.MuiDataGrid-renderingZone` was renamed `.MuiDataGrid-virtualScrollerRenderZone`
+  4. `.MuiDataGrid-gridMenuList` was renamed `.MuiDataGrid-menuList`
+  5. `.MuiGridToolbarContainer-root` was renamed `.MuiDataGrid-toolbarContainer`
+  6. `.MuiGridMenu-root` was renamed `.MuiDataGrid-menu`
+  7. `.MuiDataGridColumnsPanel-root` was renamed `.MuiDataGrid-columnsPanel`
+  8. `.MuiGridPanel-root` was renamed `.MuiDataGrid-panel`
+  9. `.MuiGridPanelContent-root` was renamed `.MuiDataGrid-panelContent`
+  10. `.MuiGridPanelFooter-root` was renamed `.MuiDataGrid-panelFooter`
+  11. `.MuiDataGridPanelHeader-root` was renamed `.MuiDataGrid-panelHeader`
+  12. `.MuiGridPanelWrapper-root` was renamed `.MuiDataGrid-panelWrapper`
+  13. `.MuiGridFilterForm-root` was renamed `.MuiDataGrid-filterForm`
+  14. `.MuiGridToolbarFilterButton-root` was renamed `.MuiDataGrid-toolbarFilterList`
```

Comment:
```suggestion
  - `.MuiDataGrid-viewport` was renamed `.MuiDataGrid-virtualScroller`
  - `.MuiDataGrid-dataContainer` was renamed `.MuiDataGrid-virtualScrollerContent`
  - `.MuiDataGrid-renderingZone` was renamed `.MuiDataGrid-virtualScrollerRenderZone`
  - `.MuiDataGrid-gridMenuList` was renamed `.MuiDataGrid-menuList`
  - `.MuiGridToolbarContainer-root` was renamed `.MuiDataGrid-toolbarContainer`
  - `.MuiGridMenu-root` was renamed `.MuiDataGrid-menu`
  - `.MuiDataGridColumnsPanel-root` was renamed `.MuiDataGrid-columnsPanel`
  - `.MuiGridPanel-root` was renamed `.MuiDataGrid-panel`
  - `.MuiGridPanelContent-root` was renamed `.MuiDataGrid-panelContent`
  - `.MuiGridPanelFooter-root` was renamed `.MuiDataGrid-panelFooter`
  - `.MuiDataGridPanelHeader-root` was renamed `.MuiDataGrid-panelHeader`
  - `.MuiGridPanelWrapper-root` was renamed `.MuiDataGrid-panelWrapper`
  - `.MuiGridFilterForm-root` was renamed `.MuiDataGrid-filterForm`
  - `.MuiGridToolbarFilterButton-root` was renamed `.MuiDataGrid-toolbarFilterList`
```

Reply:
I think we used numbers on the CHANGELOG to avoid nested lists.
I like the table version more for long lists, I'm switching to table for the selector and CSS classes lists